README.md

```diff
@@ -12,4 +26,488 @@ There is a list of all entities that presented in the project:
+- **Spring Data JPA:** Spring Data JPA simplifies database operations and enables easy interaction with the database.
+- **Swagger:** Swagger is integrated to provide interactive API documentation and testing capabilities.
+- **Mapstruct:** Mapstruct is used for object mapping between DTOs and entities.
+- **MySQL:** MySQL is used as the database management system.
+- **Liquibase:** Liquibase is employed for database version control and management.
+- **Docker:** Docker is used for containerization of the application and database.
+
+## Application API
+####
+[Back to application content](#content)
+
+1. [Authentication Controller](#authentication-controller)
+2. [Book Controller](#book-controller)
+3. [Category Controller](#category-controller)
+4. [Shopping Cart Controller](#shoppingcart-controller)
+5. [Order Controller](#order-controller)
+6. [Order Item Controller](#orderitem-controller)
+
+
+## Authentication Controller
+The Authentication Controller is responsible for user authentication and registration functionalities, enabling users to
+interact with the system and utilize its functionalities.
+- [Back to application content](#content)
+####
+
+####
+
+### User Registration:
+To register as a new user and gain access to the system's features, please follow these steps:
+
+- **POST `/api/auth/register:`** with your registration details.
+
+   - Example request body:
+   ```json
+   {
+     "email": "john.doe@example.com",
+     "password": "securePassword123",
+     "repeatPassword": "securePassword123",
+     "firstName": "John",
+     "lastName": "Doe",
+     "shippingAddress": "123 Main St, City, Country"
+   }
+  ```
+  - Example of response body:
+  ```json
+  {
+     "id": 1,
+     "email": "john.doe@example.com",
+     "firstName": "John",
+     "lastName": "Doe",
+     "shippingAddress": "123 Main St, City, Country"
+  }
+    ```
+###
+
+### User Login:
+If you are already a registered user, you can log in to access your account and perform various actions. To log in, please follow these steps:
+
+- **POST `/api/auth/login`** with your login credentials. 
+
+   - Example request body:
+   ```json
+   {
+     "email": "john.doe@example.com",
+     "password": "securePassword123"
+   }
+   ```
+  Upon successful login, you will receive an authentication token that should be included in your subsequent requests for secure access.
+    - Example response body:
+    ```json
+   {
+      "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c"
+   }
+    ```
+
+## Book Controller
+The Book Controller provides access to book-related functionalities, allowing users to retrieve and manage book information within the online store.
+- [Back to Application API content](#application-api)
+- [Back to application content](#content)
+####
+- **GET /api/books:** Retrieve the book catalog.
```

Comment:
OPTIONAL: I think it will be better to make a request samples like postman collection import JSON

Reply:
I already have instructions for Importing a Test Data Set into Postman with Postman sample data set (7 point in the application content)